### PR TITLE
fix issue with prompt's callback registration

### PR DESCRIPTION
**Pull request by Bolisov**

### CODE REVIEW

**Comment by SamVerschueren** [comment]
Can you elaborate the change a little bit? Why do you think this change is necessary?

I don't think it is to be honest as the original code made sure you could choose between promises or callbacks e.g.

```js
adapter.prompt(questions).then(() => {
     // handle result
});

adapter.prompt(questions, () => {
     // handle result
});
```

The change made means that the returned promise by the `prompt` function waits until the callback completes. With this change you can do something like this

```js
adapter
    .prompt(questions, () => 'foo')
    .then(result => {
        console.log(result);
        //=> 'foo'
    });
```

I don't think that's what you want as the the callback and promise serve the same purpose and aren't really related.

**Comment by Bolisov** [comment]
Yes, you are right. That's my fault.

File: lib/adapter.js [hunk 1]
```diff
@@ -80,6 +80,6 @@ TerminalAdapter.prototype.log = logger();
 
 TerminalAdapter.prototype.prompt = function (questions, cb) {
   var promise = this.promptModule(questions);
-  promise.then(cb || _.noop);
-  return promise;
+  var newPromise = promise.then(cb || _.noop);
+  return newPromise;
```
Comment (addyosmani) [inline lib/adapter.js]: Can you provide more justification on why this change is needed? I don't believe this specific patch does quite what you might be expecting it to (see the next comment).

Comment (Bolisov) [inline lib/adapter.js]: Yes, you are right. That's my fault.
